### DISCUSSION TITLE
Extending a custom schema

### DISCUSSION THREAD

**Discussion by ncioj10** [graphql-java-kickstart/graphql-spring-boot#513] [Q&A]
Hi,
I am using rejoiner for generating my schema which I am generating and injecting using: 
```
@Bean
 GraphQLSchema schema() {
        GraphQLSchema schema=  Guice.createInjector(
                new SchemaProviderModule(),
                anamnesisSchemaModule,
                patientSchemaModule,
                treatmentSchemaModule,
                diagnosisSchemaModule,
                furtherProceduresSchemaModule,
                treatmentLetterSchemaModule,
                physicalSchemaModule,
                knowledgeBaseSchemaModule,
                letterGeneratorBaseSchemaModule,
                userSchemaModule,
                proxySchemaModule
                )
                .getInstance(Key.get(GraphQLSchema.class, Schema.class));
return schema;
}
```

This works very well, but now I am trying to implement subscriptions, so I need to extend the given schema. 
I have tried according to the example with implementing a Resolver with `GraphQLSubscriptionResolver` interface.
This leads the library to start searching for schema file definitions in resources (which aren't there). 

How can I add subscriptions with resolvers to my schema so they are included in the generated custom schema from above?

Thank you very much in advance,
Nico

**Comment by oliemansm** [comment]
If you don't use graphql-java-tools or annotations, then maybe using graphql-java Spring library might be a better fit instead? See https://www.graphql-java.com/tutorials/getting-started-with-spring-boot/

**Comment by ncioj10** [comment]
Hi thank you, 
I already had a look at that and came up with a solution involving adding it like that:


```
        GraphQLFieldDefinition definition = GraphQLFieldDefinition.newFieldDefinition()
                .name("onTestNotification")
                .argument(GraphQLArgument.newArgument().name("userId").type(new GraphQLTypeReference("String")).build())
                .type(new GraphQLTypeReference("TestType"))
                .dataFetcher(fetcher())
                .build();


       return GraphQLSchema.newSchema(schema).subscription(GraphQLObjectType.newObject().name("Subscription").field(definition).build()).build();
```

It would be nice though if it was possible to merge Resolver based auto generated via graphql-tools subscriptions with the custom generated schema?
The problem is bascially that once I use a Interface marker from graphql-tools I can no longer inject my custom schema via the bean. 
The question is basically how to merge custom with auto generated in graphql-tools I guess?
